src/main/java/io/fusionauth/jwt/ec/ECDSASignature.java

```diff
@@ -86,10 +86,17 @@ public byte[] derDecode(Algorithm algorithm) throws IOException {
     }
 
     int len = result.length / 2;
-    //noinspection ManualMinMaxCalculation
-    System.arraycopy(r, r.length > len ? 1 : 0, result, r.length < len ? 1 : 0, r.length > len ? len : r.length);
-    //noinspection ManualMinMaxCalculation
-    System.arraycopy(s, s.length > len ? 1 : 0, result, s.length < len ? (len + 1) : len, s.length > len ? len : s.length);
+
+    int rSrcPos = r.length > len ? (r.length - len) : 0;
+    int rDstPos = Math.max(0, len - r.length);
+    int rLength = Math.min(r.length, len);
+    System.arraycopy(r, rSrcPos, result, rDstPos, rLength);
```

Comment:
Looking at these more, I think that both `r` and `s` only need the `dstPos` adjusted.
- The `srcPos` will always be `0` for both values
- The `length` will always `r.length` or `s.length`
- `rDstPos` is always `len - r.length`
- `sDstPos` is always `len + (len - s.length)` or `2 * len - s.length`

Reply:
The incoming value can also be padded, so the `srcPos` won't always be `0`, at least not in theory according to the encoding spec. 

We can discuss it further if you'd like.